Create hacs.json

rsnodgrass opened the pull request:
Add configuration to make easily installation and automatic upgrading using HACS (Home Assistant Community Store).

rsnodgrass commented:
Yes.    Technically they need to be in the repo under custom_components, rather than a noaa_tides at the top level.    I forgot about that aspect and seeing if there is a way to override that behavior (I hope since it seems like HACS should not put too many requirements on existing structure).    
  

  
I had to move it into custom_components/noaa_tides in my git fork to make it work so far.   
  

  
  

  
  
>   
> On Jul 18, 2020 at 1:01 PM,  <Jacob Shufro (mailto:notifications@github.com)>  wrote:
>   
>   
>   
>
>
>
>   
>
> @jshufro  commented on this pull request.
>
>   
>
> Any idea how this will behave in conjunction with the existing NOAA Tides and Current sensor available in core? Does HACS just drop the files in custom_components?
>
>   
>
> —
>  You are receiving this because you authored the thread.
>  Reply to this email directly,   view it on GitHub (https://github.com/jshufro/home_assistant_noaa_tides/pull/7#pullrequestreview-451079241), or   unsubscribe (https://github.com/notifications/unsubscribe-auth/AAQY4XDVVWWKMXFR6OKE3QTR4H5TNANCNFSM4PAFR4QA).
>
>   
>

jshufro commented:
If you want to put a symlink in named custom_components/noaa_tides and point it at the existing folder, that may appease HACs without having to push a breaking change.

rsnodgrass commented:
After testing, unfortunately HACS does not work with a symlink.

I have updated by moving noaa_tides to a custom_components folder, and updated the README.md to reflect the new location.  Is that ok?  If so, I will submit a NEW pull request that includes this, documentation updates, and one bug fix for an error message.

jshufro commented:
Sure. It's a shame HACS doesn't work with symlinks but I am guessing it's because of limitations around how they validate a project before cloning it.
